packages/contracts/contracts/registryManager/RegistryManager.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+import { Common } from "../common/Common.sol";
+import { IRecipientRegistry } from "../interfaces/IRecipientRegistry.sol";
+import { IRegistryManager } from "../interfaces/IRegistryManager.sol";
+
+/// @title RegistryManager
+/// @notice Contract that allows to use send, approve, reject requests to RecipientRegistry.
+contract RegistryManager is Ownable, IRegistryManager, Common {
+  /// @notice requests
+  mapping(uint256 => Request) internal requests;
+
+  /// @notice request statuses
+  mapping(uint256 => Status) internal statuses;
+
+  /// @inheritdoc IRegistryManager
+  uint256 public requestCount;
+
+  /// @notice Initialize registry manager
+  constructor() payable Ownable(msg.sender) {}
+
+  /// @notice Check if request is valid
+  modifier isValidRequest(Request calldata request) {
+    // TODO: check registry owner as well to make sure it's a part of maci
+    if (request.registry == address(0)) {
+      revert ValidationError();
+    }
+
+    if (request.recipient.recipient == address(0)) {
+      revert ValidationError();
+    }
+
+    uint256 count = IRecipientRegistry(request.registry).recipientCount();
+
+    if (request.index >= count && request.requestType == RequestType.Change) {
+      revert ValidationError();
+    }
+
+    _;
+  }
+
+  /// @notice Check if request is pending and exists
+  /// @param index Request index
+  modifier isPending(uint256 index) {
+    if (index >= requestCount || statuses[index] != Status.Pending) {
+      revert OperationError();
+    }
+    _;
+  }
+
+  /// @inheritdoc IRegistryManager
+  function send(Request calldata request) public virtual override isValidRequest(request) {
+    requests[requestCount] = request;
+    statuses[requestCount] = Status.Pending;
+
+    unchecked {
+      requestCount++;
+    }
+
+    emit RequestSent(
+      request.registry,
+      request.requestType,
+      request.recipient.recipient,
+      request.index,
+      request.recipient.id,
+      request.recipient.metadataUrl
+    );
+  }
+
+  /// @inheritdoc IRegistryManager
+  function approve(uint256 index) public virtual override onlyOwner isPending(index) {
+    Request memory request = requests[index];
+    IRecipientRegistry registry = IRecipientRegistry(request.registry);
+
+    if (request.requestType == RequestType.Change) {
+      approveRequest(index, request);
+      registry.changeRecipient(request.index, request.recipient);
+    } else {
+      approveRequest(index, request);
+      registry.addRecipient(request.recipient);
+    }
+  }
+
+  /// @notice Internal approve request with state change
+  /// @param index Index of the request (optional)
+  /// @param request Request to the registry
+  function approveRequest(uint256 index, Request memory request) internal {
```

Comment:
should we go with `_` prefix for internal functions?

Reply:
Linter doesn't mind if there is no underscore for internal functions.